### PR TITLE
Feature/version 2.0

### PR DESCRIPTION
**Pull request by tomme87**
Updated to version 2 of bundle.

- Request entities are no longer created using reflection, but we are serializing
- Request needs to be validated before creation.
  - All request must implement a method that returns to constraint to validate the request data with.
- Request entity options is now return by the request, not in paramconverter options.
- JsonRequestSubscriber moved to this bundle (can be turn off with decode_json_requests config option)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,3 +43,111 @@ return [
...
+           'allowMissingFields' => static::allowMissingFields(),
+           'allowExtraFields' => static::allowExtraFields()
+         ])
+       ];
+     }
+   
+   }
+   ```
+   Override the `getRequestDataValidationFields` method to validate the request data.
+   These validation rules must match the incoming request which is an array (i.e after decoding a json request)
```
Comment (marco-giannetti) [inline README.md]: I think it should be i.e.

File: README.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 HalloVerdenRequestEntityBundle
 ==============================
+Deserializes a request to a entity.
```
Comment (marco-giannetti) [inline README.md]: Should be _an_ entity?

**Review by marco-giannetti — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (tomme87) [reply README.md]: Fixed (b04a8a0)

File: README.md (hunk 2 above)
Comment (tomme87) [reply README.md]: Fixed (484ac76)

**Review by erichjsonfosse — APPROVED** [review]

**Pull request merged**